server/src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -72,6 +72,7 @@ private static void executeInSortOrder(SearchContext context, BucketCollector co
         searcher.setProfiler(context);
         try {
             searcher.search(context.rewrittenQuery(), collector);
+            collector.postCollection();
```

Comment:
I am thinking that this is another exception to the offloading that we should remove. We should provide an executor to the time series index searcher, and offload its sequential execution to it.

Reply:
I was thinking the same thing, when making that change.
However this isn't a normal searcher, but a `TimeSeriesIndexSearcher` and does searching completely different.

Reply:
yep, I think though that because it's part of the query phase, it could potentially cause us to take in more heavy work than desired, say that we have many searches being offloaded and at the same time many others using this time series index searcher, we'd have many threads (potentially double compared to before adding the search workers thread pool) doing I/O and CPU bound stuff.

I don't think this is a blocker for this PR but we should look into it as a follow-up.